financial-connections/src/main/java/com/stripe/android/financialconnections/FinancialConnectionsSheetViewModel.kt

```diff
@@ -83,7 +92,7 @@ internal class FinancialConnectionsSheetViewModel @Inject constructor(
         setState {
```

Comment:
shouldn't we always wrap set state in mutex lock?

Reply:
We just need to do it for the lifecycle events that get triggered immediately one after the other so that we avoid concurrency issues. That is:

- onResume: every time the SDK activity goes to background (like when opening the web browser)
- onNewIntent: when coming back from the web browser / bank app with a deeplink
- onActivityResult: when the browser / bank app activity is closed.

We rely on this 3 callbacks to determine the status of the external (web / app2app) flow. 

`openAuthFlow` is not part of the lifecycle, so we should be fine not adding the mutex here : )